conf/src/lib.rs

```diff
@@ -104,6 +104,18 @@ impl CLNConf {
         Ok(())
     }
 
+    pub fn rm_conf(&mut self, key: &str) -> Result<(), ParsingError> {
+        if self.fields.contains_key(key) {
+            self.fields.remove_entry(key);
+        } else {
+            return Err(ParsingError {
+                core: 2,
+                cause: format!("field with `{key}` not present"),
+            });
+        }
+        Ok(())
+    }
+
```

Comment:
```suggestion
    pub fn rm_conf(&mut self, key: &str, value: Option<&str>) -> Result<(), ParsingError> {
        if self.fields.contains_key(key) {
             // 1. get the value from the map
             // 2. is the value is not known see if the value is equal to the previous 
            self.fields.remove_entry(key);
        } else {
            return Err(ParsingError {
                core: 2,
                cause: format!("field with `{key}` not present"),
            });
        }
        Ok(())
    }

```

OK I found a bug, the filed in the config file are stored as <Key, Vec<Values>> this mean that for a `plugin` field you have a single entry inside the hash map but multiple value.

The quest is we want remove all the entries with a `plugin` field? the answer is:
- Yes when the `value` from the parameter is None;
- No, we want remove a specific value with a parameter with `value` that the user specified

This is useful in your case in coffee, you want remove a plugin and this mean that the plugin is formed by key=plugin, value=path of the plugin

Reply:
Yes. I didn't notice.
Thanks for the review.